deploy/roles/spark/tasks/main.yml

```diff
@@ -0,0 +1,91 @@
+---
+- name: Create service account for Spark
+  user: 
+    name: "{{ spark_user }}"
+    system: yes
+    home: "{{ spark_lib_dir }}"
+    shell: "{{ spark_user_shell }}"
+    state: present
+    groups: "{{ spark_user_groups | join(',') }}"
+    comment: "Spark-User"
+
+- name: Make sure Spark configuration directory exists
+  file: 
+    path: "{{ spark_conf_dir }}"
+    state: directory
+
+- name: Make sure Spark directories exists
+  file: 
+    path: "{{ item }}" 
+    owner: "{{ spark_user }}"
+    group: "{{ spark_user }}"
+    mode: 0755, a+x
+    state: directory
+  with_items:
+    - "{{ spark_run_dir }}"
+    - "{{ spark_main_dir }}"
+
+- name: Ensure Spark log directory with appropriate permissions
+  file: 
+    path: "{{ item }}" 
+    owner: "{{ spark_user }}"
+    group: "{{ spark_user }}"
+    mode: 1777, a+x
```

Comment:
Same here, remove the a+x, and let's make it 1755 if that works (or 1775 if necessary)